lib/rubocop/cop/rspec/variable_name.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # This cop makes sure that all variables use the configured style.
+      #
+      # @example EnforcedStyle: snake_case (default)
+      #   # bad
+      #   let(:userName) { 'Adam' }
+      #   subject(:userName) { 'Adam' }
+      #
+      #   # good
+      #   let(:user_name) { 'Adam' }
+      #   subject(:user_name) { 'Adam' }
+      #
+      # @example EnforcedStyle: camelCase
+      #   # bad
+      #   let(:user_name) { 'Adam' }
+      #   subject(:user_name) { 'Adam' }
+      #
+      #   # good
+      #   let(:userName) { 'Adam' }
+      #   subject(:userName) { 'Adam' }
+      class VariableName < Cop
+        include RuboCop::RSpec::Language::NodePattern
+        include ConfigurableNaming
+
+        MSG = 'Use %<style>s for variable names.'
+
+        def_node_matcher :variable_definition?, <<~PATTERN
+          (send #{RSPEC} #{Variables::ALL.node_pattern_union}
+            ({sym str} $_) ...)
```

Comment:
perhaps we need a cop to ensure let/subject are being used with a symbol and not a string?

Reply:
I created an issue for this: https://github.com/rubocop-hq/rubocop-rspec/issues/908